timesketch/lib/analyzers/ioc_extraction.py

```diff
@@ -0,0 +1,280 @@
+        super(IOCExtractionSketchPlugin, self).__init__(
+            index_name, sketch_id)
+        self._config = config
+
+    def run(self):
+        """Entry point for the analyzer.
+
+        Returns:
+            String with summary of the analyzer result.
+        """
+        config = self._config or interface.get_yaml_config(self.CONFIG_FILE)
+        if not config:
+            return 'Unable to parse the config file.'
+        query, ret_fields, ioc_db = self.parse_config(config)
+        if not query or not ret_fields:
+            return 'Error to make query'
+        if not ioc_db:
+            return 'Error to parse config'
+        logger.info((
+            'Return field '
+            'error: {0!s}').format(ret_fields))
+        return self.extract_ioc(query, ret_fields, ioc_db)
+
+    def parse_config(self, ioc_config):
+        """Create IOC DB and make request
+
+        Args:
+            config: A dict that contains the configuration for the ioc
+                extraction. See data/ioc.yaml for fields and further
+                documentation of what needs to be defined.
+
+        Returns:
+            query: A string that contains elastic query to return
+                events check ioc against.
+            return_fields: A list that contains names of field to return
+                from elastic request to check ioc against.
+            ioc_db: A dict contains extracted config and IOC files
+        """
+        ioc_db = []
+        query = ""
+        return_fields = []
+        list_fieldnames = self.get_fields_list()
+        for name, config in iter(ioc_config.items()):
+            elm = {'name': name, 'found': 0}
+            path_file_ioc = config.get('path_file_ioc')
+            if not path_file_ioc:
+                logger.warning('No path for file IOC defined.')
+                continue
+            if not os.path.isfile(path_file_ioc):
+                logger.warning('file IOC defined not exists.')
+                return ''
+            elm['ioc'] = None
+            with open(path_file_ioc) as json_file:
+                try:
+                    elm['ioc'] = json.load(json_file)
+                    if not isinstance(elm['ioc'], (list, tuple)):
+                        continue
+                    if not elm['ioc']:
+                        continue
+                except json.JSONDecodeError as exception:
+                    logger.warning((
+                        'File IOC error to parse: '
+                        '{0!s}').format(exception))
+                    continue
+            elm['store_as'] = config.get('store_as')
+            if not elm['store_as']:
+                logger.warning('No attribute defined to store results in.')
+                continue
+            elm['attributes'] = config.get('attributes')
+            attri_contains = config.get('attributes_contains')
+            if attri_contains:
+                for fn in list_fieldnames:
```

Comment:
ok, here you can just use a list comprehension...

this looks like to me that you have a list of field names... and you say

for each of the fields in a list of all the field names, go and check each of the attribute in the attribute container and check if it exists in the field... so if you've got:

```
list_fieldnames = ['apple', 'banana', 'grape']
attribute_container = ['applepie', 'ap', 'grape']
```

Then you do

```
fn = 'apple'
element['attributes'] = [
  'applepie' in 'apple' -> no
 'ap' in 'apple' -> YES
 'grape' in 'apple' -> No]

after first round
element['attributes'] = ['ap']

fn = 'banana'
element['attributes'] = [
  'applepie' in 'banana' -> no
 'ap' in 'banana' -> no
 'grape' in 'banana' -> No]

after second round
element['attributes'] = ['ap']

fn = 'grape'
element['attributes'] = [
  'applepie' in 'grape' -> no
 'ap' in 'grape' -> no
 'grape' in 'grape' -> yes]

final results

elment['attributes'] = ['ap', 'grape']
```

Is this the intention? Also seems very slow to go through these iterations, what about if list field names returns 60 fields (could very easily do so)??

or is the intention just to see if there is an exact match?

```
element['attributes'] = [x for x in attribute_container if x in list_fieldnames]
```

and then you don't have to have the `+=` which BTW is a list concatenation, which means for each time you call that loop you create a new list, leaving the old list behind in memory... so you 

list_a = []
list_a_ = []
create a new list with ['a']
overwrite list_a with the new list

...

instead of using 'append' or 'extend' on the list.